lte/gateway/c/oai/tasks/amf/amf_recv.cpp

```diff
@@ -71,51 +74,143 @@ int amf_handle_registration_request(
   } else {
     params->m5gsregistrationtype = AMF_REGISTRATION_TYPE_INITIAL;
   }
-  /* This is SUCI message identity type is SUPI as IMSI type
-   * Extract the SUPI from SUCI directly as scheme is NULL */
-  if (msg->m5gs_mobile_identity.mobile_identity.imsi.type_of_identity ==
-      M5GSMobileIdentityMsg_IMSI) {
-    // Only considering protection scheme as NULL else return error.
-    if (msg->m5gs_mobile_identity.mobile_identity.imsi.protect_schm_id ==
-        MOBILE_IDENTITY_PROTECTION_SCHEME_NULL) {
-      /*
-       * Extract the SUPI or IMSI from SUCI as scheme output is not encrypted
-       */
-      params->imsi = new (imsi_t);
+
+  ue_m5gmm_context_s* ue_context = amf_ue_context_exists_amf_ue_ngap_id(ue_id);
+  if (ue_context == NULL) {
+    OAILOG_INFO(LOG_AMF_APP, "AMF_TEST: ue_context is NULL\n");
+  }
```

Comment:
This does not return an error and if null, ut will lead to segmentation fault in the other sections of this function as you are accessing ue_context.